core/src/main/java/org/apache/gravitino/authorization/PermissionManager.java

```diff
@@ -67,7 +69,10 @@ User grantRolesToUser(String metalake, List<String> roles, String user) {
     try {
       List<RoleEntity> roleEntitiesToGrant = Lists.newArrayList();
       for (String role : roles) {
-        roleEntitiesToGrant.add(roleManager.getRole(metalake, role));
+        TreeLockUtils.doWithTreeLock(
+            AuthorizationUtils.ofRole(metalake, role),
+            LockType.READ,
+            () -> roleEntitiesToGrant.add(roleManager.getRole(metalake, role)));
```

Comment:
I guess with the overhead of requesting multiple read lock, the performance may not be as good as request one parent read lock. I would be inclined to use the old way, what do you think?

Reply:
Locking the parent node with read type is NOT correct logically. For example, if we lock the parent lock with a read lock, other threads can still modify the role when this method is loading the role, so if we are going to use the namespace lock, the lock type should be changed to `WRITE`. Is that acceptable to you?